grpc-sys/grpc_wrap.c

```diff
@@ -76,10 +76,17 @@
 #define GPR_CALLTYPE
 #endif
 
+grpc_slice grpc_slice_from_buffer(const char *source, size_t length) {
+    if (length == 0) return grpc_empty_slice();
+    grpc_slice slice = grpc_slice_malloc_large(length);
```

Comment:
why do we need to malloc?

Reply:
gRPC's slice uses either rc gc or copy semantics, I was still investigating about how to remove this malloc (cus this malloc the memories for the rc functions' vtable as well, I can create a fake rc to replace the built-in one)